KEDamage.cs

```diff
@@ -8,58 +8,73 @@ class KEDamage:Damage
     {
         public struct KineticData
         {
+            /// <summary>
+            /// The pierce of the shot
+            /// </summary>
             public float Pierce;
-
+            /// <summary>
+            /// Distance between the units
+            /// </summary>
             public float Distance;
-
-            public float ArmorKEFactor;
-
-            public float HealthKEFactor;
-
+            /// <summary>
+            /// Multiplier for armor degradation
+            /// </summary>
+            public float ArmorDegradationFactor;
+            /// <summary>
+            /// Multiplier for health damage
+            /// </summary>
+            public float HealthDamageFactor;
+            /// <summary>
+            /// Air friction constant used in calculation of attenuation
+            /// </summary>
             public float FrictionFactor;
         }
 
-        private KineticData data;
-        private Damage.Target target;
+        private KineticData _keData;
 
-        public KEDamage(KineticData d,Damage.Target t):base(EDamageTypes.KE)
+        public KEDamage(KineticData d,Damage.Target t):base(EDamageTypes.KE, t)
         {
-            data = d;
-            target = t;
+            _keData = d;
         }
 
-        override public float DealDamage()
+        public override Target CalculateDamage()
         {
+            Target finalState = _target;
+
             // Calculate attenuation of air friction
-            data.Pierce = CalculateKEAttenuationSimple(data.Pierce,data.Distance,data.FrictionFactor);
+            _keData.Pierce = CalculateKEAttenuationSimple(_keData.Pierce,_keData.Distance,_keData.FrictionFactor);
 
             // Calculate effects of ERA
-            data.Pierce = CalculatePostERAPierce(data.Pierce,target.ERAData.KEFractionMultiplier);
+            float finalERA = Math.Max(0.0f, _target.ERAData.CurrentValue
+             - _keData.Pierce*_target.ERAData.KEFractionMultiplier);
```

Comment:
`float finalERA = Math.Max(0f, _target.ERAData.CurrentValue - 
                               _keData.Pierce * _target.ERAData.KEFractionMultiplier);`
Space out them math-fermulers'. 
You also can just use 0f isntead of 0.0f but that's just preference.